src/core/net/ip6.cpp

```diff
@@ -414,7 +414,7 @@ ThreadError Ip6::HandleDatagram(Message &message, Netif *netif, uint8_t interfac
 
 exit:
 
-    if (error == kThreadError_Drop)
+    if ((error == kThreadError_Drop) || (error == kThreadError_NoRoute))
```

Comment:
Couple questions:

1. When does `error` get set to kThreadError_NoRoute in this method?
2. If we want this to be a general catch-all, should we just change the if-statement to be `error != kThreadError_None`?